clang 3.9.1 Wundefined-var-template warnings

I just installed clang 3.9.1 and I am getting a ton of warnings of the following kind:
```
/ssd/deal-git/source/numerics/error_estimator_1d.cc:94:41: warning: instantiation of variable 'dealii::StaticMappingQ1<1, 1>::mapping' required here, but no definition is available [-Wundefined-var-template]
  estimate(StaticMappingQ1<1,spacedim>::mapping, dof_handler, quadrature, neumann_bc, solution,
                                        ^
/ssd/deal-git/include/deal.II/fe/mapping_q1.h:93:41: note: forward declaration of template entity is here
  static MappingQGeneric<dim, spacedim> mapping;
                                        ^
/ssd/deal-git/source/numerics/error_estimator_1d.cc:94:41: note: add an explicit instantiation declaration to suppress this warning if 'dealii::StaticMappingQ1<1, 1>::mapping' is explicitly instantiated in another translation unit
  estimate(StaticMappingQ1<1,spacedim>::mapping, dof_handler, quadrature, neumann_bc, solution,
                                        ^
```
I will investigate later today.

This is perfectly valid C++ code except that linking might fail because a definition of `Foo<t>::foo` is not available in the translation unit. gcc happily compiles and fails to link:
```
% g++ -Wall test.cc
...
/tmp/ccRdXTao.o: In function `Foo<int>::function()':
test.cc:(.text._ZN3FooIiE8functionEv[_ZN3FooIiE8functionEv]+0x2b): undefined reference to `Foo<int>::foo'
...
```

Whereas clang emits a warning that something might go wrong:
```
% clang++ test.cc  
test.cc:6:13: warning: instantiation of variable 'Foo<int>::foo' required here, but no definition is available [-Wundefined-var-template]
    return *foo;
            ^
test.cc:18:9: note: in instantiation of member function 'Foo<int>::function' requested here
    foo.function();
        ^
test.cc:26:12: note: in instantiation of member function 'Bar<int>::Bar' requested here
  Bar<int> bar;
           ^
test.cc:10:15: note: forward declaration of template entity is here
  static int *foo;
              ^
1 warning generated.
/tmp/test-6d50ac.o: In function `Foo<int>::function()':
test.cc:(.text._ZN3FooIiE8functionEv[_ZN3FooIiE8functionEv]+0x8): undefined reference to `Foo<int>::foo'
clang-3.9: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/cmake/setup_compiler_flags_gnu.cmake
+++ b/cmake/setup_compiler_flags_gnu.cmake
@@ -111,6 +111,12 @@ IF(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-c++11-extensions")
 
   #
+  # Disable a diagnostic that warns about potentially uninstantiated static
+  # members. This leads to a ton of false positives.
+  #
+  ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-undefined-var-template")
+
+  #
   # Clang versions prior to 3.6 emit a lot of false positives wrt
   # "-Wunused-function". Also suppress warnings for Xcode older than 6.3
   # (which is equivalent to clang < 3.6).

